client/components/domains/use-your-domain-step/index.tsx

```diff
@@ -0,0 +1,316 @@
+import { isPlan } from '@automattic/calypso-products';
+import page from '@automattic/calypso-router';
+import { Card, Button, Gridicon } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import {
+	CALYPSO_HELP_WITH_HELP_CENTER,
+	INCOMING_DOMAIN_TRANSFER,
+	MAP_EXISTING_DOMAIN,
+} from '@automattic/urls';
+import { formatCurrency, useTranslate } from 'i18n-calypso';
+import { get, isEmpty } from 'lodash';
+import { stringify } from 'qs';
+import React, { useState } from 'react';
+import migratingHostImage from 'calypso/assets/images/illustrations/migrating-host-diy.svg';
+import themesImage from 'calypso/assets/images/illustrations/themes.svg';
+import QueryProducts from 'calypso/components/data/query-products-list';
+import HeaderCake from 'calypso/components/header-cake';
+import {
+	isDomainBundledWithPlan,
+	isDomainMappingFree,
+	isNextDomainFree,
+} from 'calypso/lib/cart-values/cart-items';
+import {
+	getDomainPrice,
+	getDomainProductSlug,
+	getDomainTransferSalePrice,
+} from 'calypso/lib/domains';
+import useCartKey from 'calypso/my-sites/checkout/use-cart-key';
+import { useSelector, useDispatch } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import {
+	DOMAINS_WITH_PLANS_ONLY,
+	NON_PRIMARY_DOMAINS_TO_FREE_USERS,
+} from 'calypso/state/current-user/constants';
+import { currentUserHasFlag } from 'calypso/state/current-user/selectors';
+import { getProductsList } from 'calypso/state/products-list/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+import './style.scss';
+
+const ANALYTICS_SECTION = 'domains';
+
+const PAGE_URL = '/use-your-domain';
+function extractBasePath( basePath: string ) {
+	return basePath.endsWith( PAGE_URL )
+		? basePath.substring( 0, basePath.length - PAGE_URL.length )
+		: basePath;
+}
+
+type OptionContentProps = {
+	image: string;
+	title: string;
+	reasons: React.ReactNode[];
+	onClick: React.MouseEventHandler;
+	buttonText: string;
+	isPrimary: boolean;
+	learnMore: React.ReactNode;
+};
+
+function UseYourDomainStepContent( {
+	image,
+	title,
+	reasons,
+	onClick,
+	buttonText,
+	isPrimary,
+	learnMore,
+}: OptionContentProps ) {
+	return (
+		<Card className="use-your-domain-step__option" compact>
+			<div className="use-your-domain-step__option-inner-wrap">
+				<div className="use-your-domain-step__option-content">
+					<div className="use-your-domain-step__option-illustration">
+						<img src={ image } alt="" />
+					</div>
+					<h3 className="use-your-domain-step__option-title">{ title }</h3>
+					<div className="use-your-domain-step__option-reasons">
+						{ reasons.map( ( phrase, index ) => {
+							if ( isEmpty( phrase ) ) {
+								return;
+							}
```

Comment:
This `isEmpty` call be eliminated in favor of `phrase === null`. The value (member of `reason` array) is always either `null` or a result of `translate()`.

Reply:
https://github.com/Automattic/wp-calypso/pull/101854/commits/f52f737944224af3059dfe29f5ad3d8ff3c5dcdb